include/swift/Runtime/Distributed.h

```diff
@@ -0,0 +1,68 @@
+//===--- Distributed.h - Runtime interface for distributed ------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// The runtime interface for distributed.
+//
+//===----------------------------------------------------------------------===//
```

Comment:
Very happy to move stuff into it's own header finally <3

Reply:
I mean this state of the PR is a very rough WIP I shared to get help on getting unblocked...

My intent was to get your changes merged or cherry pick them here, I'm not worried about the conflicts to be honest -- happy and easy to resolve them